string robot
string location
string can
---
string result